WooCommerce/Classes/ViewRelated/Orders/Payment Methods/Cash/CashPaymentTenderViewModel.swift

```diff
@@ -44,7 +44,7 @@ final class CashPaymentTenderViewModel: ObservableObject {
         self.onOrderPaid = onOrderPaid
         self.analytics = analytics
         self.currencyFormatter = .init(currencySettings: storeCurrencySettings)
-        formattableAmountViewModel.presetAmount(formattedTotal)
+        formattableAmountViewModel.presetAmount(currencyFormatter.convertToDecimal(total)?.decimalValue ?? 0)
```

Comment:
Good question. 

We can always log a failure to convert. Even within the currency formatter itself, so we would see something in the logs if users report issues.

In this instance, the conversion to `Decimal` can fail if `CashPaymentTenderViewModel` is initialized with `total: String` in an unexpected format. And technically this value is completely unconstrained. Anything can be passed through a `total: String`. It would be impossible to unit test this. A safer approach would be to always pass these values as `Decimal` rather than `String`, and only turn between `Decimal` and `String` in two cases:

1. When the value comes from the API
2. When the user inputs the value

I'll check what it would mean passing `Decimal` to `CashPaymentTenderViewModel` and if a conversion in a different place would be safer. However, I imagine a better solution would require a larger refactoring.

Reply:
Okay, so `CurrencyFormatter` already logs an error:

`DDLogError("Error: string input is not a number: \(stringValue)")`

I checked the chain. `total` value comes from `Order`, which _should_ be supported by `CurrencyFormatter`. Failure can only be unexpected.